app/controllers/taggings_controller.rb

```diff
@@ -15,32 +15,22 @@ def find_by_slug
   end
 
   def show
-    @content_item = ContentItem.find!(params[:content_id])
-    @tagging_update = TaggingUpdateForm.from_content_item_links(@content_item.link_set)
-    @tag_types = @content_item.allowed_tag_types
-    @linkables = Linkables.new
+    content_item = ContentItem.find!(params[:content_id])
+    @tagging_update = Tagging::TaggingUpdateForm.from_content_item(content_item)
   rescue ContentItem::ItemNotFoundError
     render "item_not_found", status: 404
   end
 
   def update
-    tagging_update_form = TaggingUpdateForm.new(params[:tagging_update_form])
     content_item = ContentItem.find!(params[:content_id])
-    tag_types = content_item.allowed_tag_types
-
-    if tagging_update_form.valid?
-      Services.publishing_api.patch_links(
-        tagging_update_form.content_id,
-        links: tagging_update_form.links_payload(tag_types),
-        previous_version: tagging_update_form.previous_version.to_i,
-      )
+    tagging_update = Tagging::TaggingUpdatePublisher.new(content_item, params[:tagging_tagging_update_form])
```

Comment:
That's actually correct - now that that class lives inside the `Tagging` module, Rails generates this param name for it.